charts/publisher/templates/worker/deployment.yaml

```diff
@@ -35,7 +35,13 @@ spec:
             - secretRef:
                 name: {{ .Release.Name }}-env
           env:
-            {{- .Values.extraEnv | toYaml | nindent 12 }}
+          {{- range $var_name, $secret := .Values.service.secrets }}
```

Comment:
I see where you're coming from in reformatting the valueFrom/secretKeyRef stuff like this, but I'm not convinced it's worth the complexity of introducing a new format and transforming it in template code, vs. just passing in the spec YAML directly in its native form.

The extra layer of string-template stuff just means we're passing stuff like this in the values files:
```
  secrets:
    GDS_SSO_OAUTH_ID:
      name: signon-app-publisher-eks
      key: oauth_id
    GDS_SSO_OAUTH_SECRET:
      name: signon-app-publisher-eks
      key: oauth_secret
```

versus this, without the templating:

```
          env:
            - name: GDS_SSO_OAUTH_ID
              valueFrom:
                secretKeyRef:
                  name: signon-app-publisher-eks
                  key: oauth_id
            - name: GDS_SSO_OAUTH_SECRET
              valueFrom:
                secretKeyRef:
                  name: signon-app-publisher-eks
                  key: oauth_secret
```

So might as well just pass the second format verbatim and avoid the string-template-programming.

Moreover, I think we can avoid the hassle of passing this stuff in via values in the first place. The names of the secrets are really just part of the Deployment's interface, so it's reasonable to just define this stuff directly in the Deployment and not let the user change the names of the secrets. Don't worry about that for now though unless it makes things easier for this PR — I get that you didn't intend to fix up all the secrets stuff as part of this.

Reply:
Consider this all optional btw - I'm happy to fix up the secrets stuff after as it's the next thing I need to work on anyway.